staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -640,3 +683,29 @@ func (in crdStorageMap) clone() crdStorageMap {
 	}
 	return out
 }
+
+// crdConversionRESTOptionsGetter overrides the codec with one using the custom converter
+// and custom encoder and decoder version.
+type crdConversionRESTOptionsGetter struct {
+	delegate       generic.RESTOptionsGetter
+	converter      runtime.ObjectConvertor
+	encoderVersion schema.GroupVersion
+	decoderVersion schema.GroupVersion
+}
+
+func (t crdConversionRESTOptionsGetter) GetRESTOptions(resource schema.GroupResource) (generic.RESTOptions, error) {
+	ret, err := t.delegate.GetRESTOptions(resource)
+	if err == nil {
+		ret.StorageConfig.Codec = versioning.NewCodec(
+			ret.StorageConfig.Codec,
+			ret.StorageConfig.Codec,
+			t.converter,
+			&unstructuredCreator{},
+			discovery.NewUnstructuredObjectTyper(nil),
```

Comment:
reminder to myself, I promised to move this to the extensions apiserver because it's the only consumer and it's misnamed.

Reply:
We might want to move all of these into a scheme package here.